core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala

```diff
@@ -379,6 +380,27 @@ abstract class EndToEndAuthorizationTest extends IntegrationTestHarness with Sas
     assertEquals(1, describeResults2.get(topic2).get().partitions().size())
   }
 
+  @Test
+  def testSendListOffsetBeforeReauthentication(): Unit = {
```

Comment:
Does this test hit the issue? If it doesn't reliably test the issue, it may be better to add a unit test since this is expensive and one of the other tests occasionally runs into this issue anyway. If it does reliably test the issue, we should keep this.

Reply:
It fails consistently on my local but you are right that it is too expensive. Will update the test case.